File: Showcase/Showcase/Importer.cs
using PlayNXNA;
using playn.core;

namespace Showcase
{
    public class Importer
    {
        public static void Import()
        {
            XNAGraphics graphics = (XNAGraphics)PlayN.graphics();
            graphics.registerFont("Helvetica", "Helvetica-24", 24, Font.Style.PLAIN);
            graphics.registerFont("Helvetica", "Helvetica-24-Bold", 24, Font.Style.BOLD);
            graphics.registerFont("Helvetica", "Helvetica-24-Italic", 24, Font.Style.ITALIC);
            graphics.registerFont("Helvetica", "Helvetica-24-Bold-Italic", 24, Font.Style.BOLD_ITALIC);
            graphics.registerFont("Helvetica", "Helvetica-12", 12, Font.Style.PLAIN);
            graphics.registerFont("Helvetica", "Helvetica-12-Bold", 12, Font.Style.BOLD);
            graphics.registerFont("Helvetica", "Helvetica-12-Italic", 12, Font.Style.ITALIC);
            graphics.registerFont("Helvetica", "Helvetica-12-Bold-Italic", 12, Font.Style.BOLD_ITALIC);
            graphics.registerFont("Helvetica", "Helvetica-9", 9, Font.Style.PLAIN);
            graphics.registerFont("Museo-300", "Museo-24", 24, Font.Style.PLAIN);
            graphics.registerFont("Courier", "CourierNew-24", 24, Font.Style.PLAIN);
            graphics.registerFont("Courier", "CourierNew-12", 12, Font.Style.PLAIN);
        }
    }
}

File: maven/xna-plugin/src/main/java/playn/xna/Importer.java
package playn.xna;

import java.io.PrintWriter;
import java.util.HashMap;

public class Importer implements IWritable {
	
	private XNAProject game;
	private String fontImport;
	
	public Importer(XNAProject game, Font[] fonts) {
		this.game = game;
		fontImport = "";
		for (Font font : fonts) {
			HashMap<String, String> map = new HashMap<String, String>();
			map.put("name", font.name);
			map.put("file", font.getFilename(false));
			map.put("size", "" + font.size);
			map.put("style", font.getPlayNStyle());
			fontImport += FormatUtils.format(FONT_FORMAT, map);
		}
	}
	
	public void write(PrintWriter writer) {
		HashMap<String, String> map = new HashMap<String, String>();
		map.put("game", game.name);
		map.put("fonts", fontImport);
		
		writer.print(FormatUtils.format(FORMAT, map));
	}
	
	public final static String FORMAT = 
			"using PlayNXNA;\n" + 
			"using playn.core;\n" + 
			"\n" + 
			"namespace $game$\n" + 
			"{\n" + 
			"    public class Importer\n" + 
			"    {\n" + 
			"        public static void Import()\n" + 
			"        {\n" + 
			"            XNAGraphics graphics = (XNAGraphics)PlayN.graphics();\n" + 
			"$fonts$" +
			"        }\n" + 
			"    }\n" + 
			"}\n";
	
	public final static String FONT_FORMAT = 
			"            graphics.registerFont(\"$name$\", \"fonts\\\\$file$.spritefont\", $size$f, Font.Style.$style$);\n";
}
